RecoMuon/GlobalTrackingTools/src/MuonTrackingRegionBuilder.cc

```diff
@@ -1,281 +1,278 @@
+  dphi = theNsigmaPhi*(staTrack.phiError());
+
+  // Region_Parametrizations to take into account possible L2 error matrix inconsistencies
+  double region_dEta = 0;
+  double region_dPhi = 0;
+  double eta = 0;
+  double phi = 0;
+
+  // eta, pt parametrization from MC study (circa 2009?)
+  if ( pt <= 10. ) {
+     // angular coefficients
+     float acoeff_Phi = (thePhiRegionPar2 - thePhiRegionPar1)/5;
+     float acoeff_Eta = (theEtaRegionPar2 - theEtaRegionPar1)/5;
+
+     eta = theEtaRegionPar1 + (acoeff_Eta)*(pt-5.);
+     phi = thePhiRegionPar1 + (acoeff_Phi)*(pt-5.) ;
+  }
+  // parametrization 2nd bin in pt from MC study  
+  if ( pt > 10. && pt < 100. ) {
+     eta = theEtaRegionPar2;
+     phi = thePhiRegionPar2;
+  }
+  // parametrization 3rd bin in pt from MC study
+  if ( pt >= 100. ) {
+     // angular coefficients
+     float acoeff_Phi = (thePhiRegionPar1 - thePhiRegionPar2)/900;
+     float acoeff_Eta = (theEtaRegionPar1 - theEtaRegionPar2)/900;
+
+     eta = theEtaRegionPar2 + (acoeff_Eta)*(pt-100.);
+     phi = thePhiRegionPar2 + (acoeff_Phi)*(pt-100.);
+  }
+
+  double region_dPhi1 = std::min(phi,dphi);
+  double region_dEta1 = std::min(eta,deta);
+
+  // decide to use either a parametrization or a dynamical region
+  region_dPhi = useFixedPhi ? theDeltaPhi : std::max(thePhiMin,region_dPhi1);
+  region_dEta = useFixedEta ? theDeltaEta : std::max(theEtaMin,region_dEta1);
+
+  float deltaR = theDeltaR;
+  double minPt = useFixedPt ? thePtMin : std::max(thePtMin,pt*0.6);
+
+
+  const MeasurementTrackerEvent* measurementTracker = nullptr;
+  if (!theMeasurementTrackerToken.isUninitialized()) {
+    edm::Handle<MeasurementTrackerEvent> hmte;
+    ev.getByToken(theMeasurementTrackerToken, hmte);
+    measurementTracker = hmte.product();
+  }
+
+  RectangularEtaPhiTrackingRegion* region = new RectangularEtaPhiTrackingRegion(dirVector, vertexPos,
+                                               minPt, deltaR,
+                                               deltaZ, region_dEta, region_dPhi,
+                                               theOnDemand,
+                                               thePrecise,
+                                               measurementTracker);
+
+  LogDebug("MuonTrackingRegionBuilder")<<"the region parameters are:\n"
+				       <<"\n dirVector: "<<dirVector
+				       <<"\n vertexPos: "<<vertexPos
+				       <<"\n minPt: "<<minPt
+				       <<"\n deltaR:"<<deltaR
+				       <<"\n deltaZ:"<<deltaZ
+				       <<"\n region_dEta:"<<region_dEta
+				       <<"\n region_dPhi:"<<region_dPhi
+				       <<"\n on demand parameter: "<<static_cast<int>(theOnDemand);
+  
+  return region;
+
+}
+
+void MuonTrackingRegionBuilder::fillDescriptions(edm::ParameterSetDescription& descriptions) {
+  edm::ParameterSetDescription desc;
+  desc.add<double>("EtaR_UpperLimit_Par1",0.25);
+  desc.add<double>("DeltaR",0.2);
+  desc.add<edm::InputTag>("beamSpot",edm::InputTag("hltOnlineBeamSpot"));

```

Comment:
Thanks for the suggestion.
This should be reasonably easy to implement.
I'm a bit skeptical, because we also have HI, and then probably something else for special reco is coming.

Does it really make more sense to precompile things if the system supports the py config layer. 
I suppose, HLT is special in this sense and it's easier to precompile.
Did I get the context about right?